### PR TITLE
Treasury self creation flow

### PR DESCRIPTION
**Pull request by rubycop**

### CODE REVIEW

File: instances/treasury-factory.near/aliases.mainnet.json
Comment (Megha-Dev-19) [inline instances/treasury-factory.near/aliases.mainnet.json:16]: I don't think we need all these, since the treasury-factory won't have the treasury dashboard

File: instances/treasury-factory.near/widget/components/Navbar.jsx
Comment (Megha-Dev-19) [inline instances/treasury-factory.near/widget/components/Navbar.jsx:10]: I don't think we need all these, since the self create flow won't be replicated

File: instances/treasury-factory.near/widget/components/OffCanvas.jsx
Comment (Megha-Dev-19) [inline instances/treasury-factory.near/widget/components/OffCanvas.jsx:11]: can we use the offcanvas component from the `treasury-devdao` if possible?

File: instances/treasury-factory.near/widget/components/ProfileAutocomplete.jsx
Comment (Megha-Dev-19) [inline instances/treasury-factory.near/widget/components/ProfileAutocomplete.jsx:3]: we do have a autocomplete component in [devhub](https://github.com/NEAR-DevHub/neardevhub-bos/blob/main/instances/devhub.near/widget/devhub/entity/proposal/AccountInput.jsx)

File: instances/treasury-factory.near/widget/components/create-treasury/SummaryStep.jsx
Comment (Megha-Dev-19) [inline instances/treasury-factory.near/widget/components/create-treasury/SummaryStep.jsx:3]: can we remove the log?

File: instances/treasury-factory.near/widget/lib/common.jsx
Comment (Megha-Dev-19) [inline instances/treasury-factory.near/widget/lib/common.jsx:3]: we can use all this from the `treasury-devdao`

**Review by Megha-Dev-19 — COMMENTED** [review]
I don't we need to have the same config, data and css for the self create flow, since it will be only one instance (not like the treasury dashboard)
can you also please add some screenshots or video, to display the UI, and also attach the ticket which it resolves.
thanks!

File: instances/treasury-factory.near/widget/components/NewAccountInput.jsx
Comment (Megha-Dev-19) [inline instances/treasury-factory.near/widget/components/NewAccountInput.jsx:33]: why are we doing this? can we set it to `null` or `""`

File: instances/treasury-factory.near/widget/components/create-treasury/AddMemberForm.jsx
Comment (Megha-Dev-19) [inline instances/treasury-factory.near/widget/components/create-treasury/AddMemberForm.jsx:34]: we should ideally do it like this to avoid mutatating it directly
```
setFields({
  ...fields,
  accountId: value ?? fields.accountId
});
```

File: instances/treasury-factory.near/widget/components/create-treasury/AddMemberForm.jsx
Comment (Megha-Dev-19) [inline instances/treasury-factory.near/widget/components/create-treasury/AddMemberForm.jsx:46]: same thing
```
setFields({
  ...fields,
  permissions: value ?? fields.permissions
});
```

File: instances/treasury-factory.near/widget/components/create-treasury/AddMembersStep.jsx
Comment (Megha-Dev-19) [inline instances/treasury-factory.near/widget/components/create-treasury/AddMembersStep.jsx:50]: can we remove the log if not required?

File: instances/treasury-factory.near/widget/components/create-treasury/ConfirmWalletStep.jsx
Comment (Megha-Dev-19) [inline instances/treasury-factory.near/widget/components/create-treasury/ConfirmWalletStep.jsx:15]: should we do it in useEffect in first render? to avoid calling it multiple times with change in state

File: instances/treasury-factory.near/widget/components/create-treasury/SummaryStep.jsx
Comment (Megha-Dev-19) [inline instances/treasury-factory.near/widget/components/create-treasury/SummaryStep.jsx:93]: why are we using local storage?
and would it matter if it is changed by some other [widget](https://docs.near.org/build/near-components/anatomy/web-methods#storageset)
may be we should be private storage?

**Review by Megha-Dev-19 — COMMENTED** [review]
Left few comments

File: instances/treasury-factory.near/widget/components/create-treasury/ConfirmWalletStep.jsx
Comment (rubycop) [reply instances/treasury-factory.near/widget/components/create-treasury/ConfirmWalletStep.jsx:15]: It containis Near `fetch` fn that is async. In case of first re-render it returns null.
using new Promise(...) doesn't work well here.
Also I've seen iin the codebase that this is common approach to get user balance.

Here we can omiit states and use variable (as in codebase). but it's not a true react way

**Comment by rubycop** [comment]
@race-of-sloths include

**Comment by race-of-sloths** [comment]
@rubycop Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Do you want to apply for monthly streak? Get 8+ score for a single PR this month and receive boost for race-of-sloths!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://race-of-sloths.com/images/bot-banner-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://race-of-sloths.com/images/bot-banner-light.svg">
    Shows inviting banner with latest news.
</picture>](https://race-of-sloths.com/latest-news)

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/rubycop?type=bot&pr=NEAR-DevHub/neardevhub-treasury-dashboard/102&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/rubycop?type=bot&pr=NEAR-DevHub/neardevhub-treasury-dashboard/102&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/rubycop)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/rubycop)

For maintainers:
- Score pull requests that participate in the Race of Sloths and receive a reward
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: instances/treasury-factory.near/aliases.mainnet.json
Comment (Megha-Dev-19) [inline instances/treasury-factory.near/aliases.mainnet.json:6]:
```suggestion
  "REPL_SPUTNIK_FACTORY_ACCOUNT": "sputnik-dao.near",
```

**Review by Megha-Dev-19 — APPROVED** [review]
LGTM

**Comment by rubycop** [comment]
Screenshot 2024-11-18 at 13 41 56
Screenshot 2024-11-18 at 13 42 07
Screenshot 2024-11-18 at 13 42 17
Screenshot 2024-11-18 at 13 42 33
Screenshot 2024-11-18 at 13 42 39
Screenshot 2024-11-18 at 13 42 57
Screenshot 2024-11-18 at 13 43 04
Screenshot 2024-11-18 at 13 43 11

**Pull request merged**
